modules/RoyalCommands/src/main/java/org/royaldev/royalcommands/death/Death.java

```diff
@@ -0,0 +1,188 @@
+package org.royaldev.royalcommands.death;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.royaldev.royalcommands.RoyalCommands;
+
+import java.util.List;
+import java.util.Random;
+
+import org.royaldev.royalcommands.RUtils;
+import org.royaldev.royalcommands.configuration.Configuration;
+
+public class Death {
+
+    private final RoyalCommands plugin;
+    private final Player killed;
+    private final EntityDamageEvent lastEntityDamageEvent;
+    private final DamageCause lastDamageCause;
+    private final DeathType deathType;
+    private final Random random = new Random();
+
+    public Death(final RoyalCommands instance, final PlayerDeathEvent ped) {
+        this(instance, ped.getEntity(), ped.getEntity().getLastDamageCause(), Death.getLastDamageCause(ped.getEntity()));
+    }
+
+    public Death(final RoyalCommands instance, final Player killed, final EntityDamageEvent event, final DamageCause lastDamageCause) {
+        this.plugin = instance;
+        this.killed = killed;
+        this.lastEntityDamageEvent = event;
+        this.lastDamageCause = lastDamageCause;
+        this.deathType = DeathType.getDeathTypeByEvent(this.getLastEntityDamageEvent());
+    }
+
+    private static DamageCause getLastDamageCause(final Player p) {
+        final EntityDamageEvent ede = p.getLastDamageCause();
+        return ede == null ? null : ede.getCause();
+    }
+
+    private String getMessageType() {
+        final String pullFrom;
+        switch (this.getLastDamageCause()) {
+            case BLOCK_EXPLOSION:
+                pullFrom = "blo";
```

Comment:
I know this particular code path is basically identical from RoyalDeath's implementation, but do we want to stick with strings as identifiers here? Or is that something we can change later without any real damage?

Reply:
I did think over this when first doing it, and change each of the types to match the DamageCause.
Where I got stuck is cases like _`FIRE`_ and _`FIRE_TICK`_ where we use `fir`, or _`PROJECTILE`_ where it can either be `pvp` or `mob`

Nothing stopping us from expanding the strings to be the full name (or description) 

I am open to suggestions of better ways to tackle this